app/command_groupmsg.go

```diff
@@ -0,0 +1,154 @@
+	"regexp"
+	"strings"
+
+	l4g "github.com/alecthomas/log4go"
+	"github.com/mattermost/mattermost-server/model"
+	goi18n "github.com/nicksnyder/go-i18n/i18n"
+)
+
+type groupmsgProvider struct {
+}
+
+const (
+	CMD_GROUPMSG = "groupmsg"
+)
+
+func init() {
+	RegisterCommandProvider(&groupmsgProvider{})
+}
+
+func (me *groupmsgProvider) GetTrigger() string {
+	return CMD_GROUPMSG
+}
+
+func (me *groupmsgProvider) GetCommand(T goi18n.TranslateFunc) *model.Command {
+	return &model.Command{
+		Trigger:          CMD_GROUPMSG,
+		AutoComplete:     true,
+		AutoCompleteDesc: T("api.command_groupmsg.desc"),
+		AutoCompleteHint: T("api.command_groupmsg.hint"),
+		DisplayName:      T("api.command_groupmsg.name"),
+	}
+}
+
+func (me *groupmsgProvider) DoCommand(a *App, args *model.CommandArgs, message string) *model.CommandResponse {
+	parsedMessage := ""
+	targetUsers := map[string]*model.User{}
+	targetUsersSlice := []string{args.UserId}
+	invalidUsernames := []string{}
+	teamId := ""
+
+	userExpr := "@?[^\\s,]+"
+	sepExpr := "\\s*,\\s*"
+	textExpr := "\\s*([^,].*)?"
+	messageExpr := fmt.Sprintf("^\\s*((?:%s%s)*%s)%s?$", userExpr, sepExpr, userExpr, textExpr)
+	messageRegex := regexp.MustCompile(messageExpr)
+	if !messageRegex.MatchString(message) {
+		return &model.CommandResponse{Text: args.T("api.command_groupmsg.invalid_arguments.app_error"), ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL}
+	}
+
+	matches := messageRegex.FindStringSubmatch(message)
+
+	for _, username := range strings.Split(matches[1], ",") {
+		username = strings.TrimSpace(username)
+		username = strings.TrimPrefix(username, "@")
+		if result := <-a.Srv.Store.User().GetByUsername(username); result.Err != nil {
+			invalidUsernames = append(invalidUsernames, username)
+		} else {
+			targetUser := result.Data.(*model.User)
+			_, exists := targetUsers[targetUser.Id]
+			if !exists && targetUser.Id != args.UserId {
+				targetUsers[targetUser.Id] = targetUser
+				targetUsersSlice = append(targetUsersSlice, targetUser.Id)
+			}
+		}
+	}
+
+	if len(invalidUsernames) == 1 {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.invalid_user.app_error"), "@"+invalidUsernames[0]),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+	if len(invalidUsernames) > 1 {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.invalid_users.app_error"), "@"+strings.Join(invalidUsernames, ", @")),
+			ResponseType: model.COMMAND_RESPONSE_TYPE_EPHEMERAL,
+		}
+	}
+
+	if len(matches) > 2 {
+		parsedMessage = matches[2]
+	}
+
+	if len(targetUsersSlice) == 2 {
+		return GetCommandProvider("msg").DoCommand(a, args, fmt.Sprintf("%s %s", targetUsers[targetUsersSlice[1]].Username, parsedMessage))
+	}
+
+	if len(targetUsersSlice) < model.CHANNEL_GROUP_MIN_USERS {
+		return &model.CommandResponse{
+			Text:         fmt.Sprintf(args.T("api.command_groupmsg.min_users.app_error"), model.CHANNEL_GROUP_MIN_USERS-1),
```

Comment:
Similarly with the rest of these translations, you don't need a fmt.Sprintf if you define the translation string like
```
"translation": "Group messages are limited to a minimun of {{.MinUsers}} users."
```
and call it with
```go
args.T("api.command_groupmsg.min_users.app_error", map[string]interface{"MinUsers": ...})
```